### DISCUSSION TITLE
Testing for production - HTTPS related issue.

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#2244] [Q&A]
I'm using pocketbase server in my local pc with real ip - `103.113.227.244:443` I also have a sub domain assigned against this real ip `live.algoramming.com:443`.  Everything is working fine. But the connection is insecure as http. Now I want to make it secure with https. Accordingly to #316 I have tried `./pocketbase serve --http="live.algoramming.com:80" --https="live.algoramming.com:443"` But it shows me this error `listen tcp 103.113.227.244:443: bind: cannot assign requested address`

I'm using fedora 37. Can anyone please tell me where I have done something wrong or how can I get https server?

![image](https://user-images.githubusercontent.com/49031412/230676688-309e82d4-b2ae-4c2e-b4eb-c3960720fe4d.png)

I have also port forward in my router configuration like this:
![image](https://user-images.githubusercontent.com/49031412/230676762-99a9ed57-6a7d-4fae-82b1-72b1e85ee316.png)

**Comment by ganigeorgiev** [comment]
PocketBase will try to bind to the IP that the domain is resolving. If you are running it locally make sure that your domain is mapped in your `/etc/hosts` file.

**Reply by sabikrahat** [reply]
No, I want to run the server using my real IP to be accessible from anywhere globally.

**Reply by ganigeorgiev** [reply]
@sabikrahat But you still have to map it to your local ip in the hosts file.

**Reply by sabikrahat** [reply]
I have also tried this. still the same issue. @ganigeorgiev  :(

![image](https://user-images.githubusercontent.com/49031412/230711472-ebdacb56-2836-42db-99ff-353712662575.png)

**Reply by ganigeorgiev** [reply]
@sabikrahat Try to map it to your local network ip (based on the router config screenshot that would be `192.168.10.101`).

**Reply by sabikrahat** [reply]
Same issue. @ganigeorgiev 

![image](https://user-images.githubusercontent.com/49031412/230712165-fcf0f78c-1e82-4772-8b9b-de77e3d60a4b.png)

**Reply by ganigeorgiev** [reply]
I'm not sure what could be the issue. Double check that you've mapped the correct local ip (_if you are not sure you can find it with `ifconfig`, `hostname -I`, etc._).

Another thing I guess could be if the ports are already in use by some other process like nginx, httpd, etc. (_but in this case the error message probably would be different_).

**Reply by sabikrahat** [reply]
Okay, the problem was for root access. after executing the command with `sudo` it starts the server but the problem is it throws this error when I try to view it in the browser. Can you please help me with this part? @ganigeorgiev sorry for bothering you too much.

![image](https://user-images.githubusercontent.com/49031412/230713839-cbd0539f-f6ed-460b-9514-46a3f51539a8.png)

**Reply by ganigeorgiev** [reply]
Sorry I'm not sure that I can help you since I don't have the full picture of your env setup. Try to revert the `/etc/hosts` file changes if you ended up doing any, maybe they are interfering. To ensure that your domain is properly resolving you can try first only with just the `--http` flag and see if the http :80 version is accessible.

**Reply by TMBL-DEV** [reply]
Can't you just setup a nginx proxy to your pocketbase? 

Then you can just use certbot for the SSL certificate.

**Reply by sabikrahat** [reply]
can you tell me more about this? @TMBL-DEV

**Reply by TMBL-DEV** [reply]
Il upload an example later on today. But I'm pretty sure you can figure it out on your own.

**Reply by sabikrahat** [reply]
Okay.... I'm waiting for it.

**Reply by TMBL-DEV** [reply]
well first of you would setup nginx according to this blog:
[ssl nginx]( https://www.nginx.com/blog/using-free-ssltls-certificates-from-lets-encrypt-with-nginx/
) this is a tutorial on how you can setup nginx with ssl. aka https

after that it would just be making it a reverse proxy.
Meaning that it has a server running on for example
https://some-domain.very-nice.com/

and that would proxy to a the pocketbase process that you can have running on the server. on port 8080.
[reverse proxy docs nginx](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/)

an example of how i am using a reverse proxy on my localhost with docker is this:
nginx conf



    worker_processes auto;
    pid /run/nginx.pid;
    
    events {
        worker_connections 1024;
    }
    
    http {
        upstream pocketbase {
            server vfl-pocketbase:8080;
        }
    
        server {
            listen 80;
            server_name pocketbase.some-company.localhost;
            client_max_body_size 10M;
    
            location / {
                proxy_pass http://pocketbase;
                proxy_set_header Host $host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_connect_timeout 300s;
                proxy_read_timeout 300s;
            }
        }
    }

**Reply by TMBL-DEV** [reply]
> Okay.... I'm waiting for it.

im not your slave man xD

**Reply by TMBL-DEV** [reply]
did it work @sabikrahat ?

**Reply by sabikrahat** [reply]
No I have tried several times..... Maybe I have some os related issue in my machine. Thank you for your helping :)
